### PR TITLE
Add: invoker & attachments & merge master

### PR DESCRIPTION
**Pull request by fangyincheng**
**What this PR does**:

  1. move callService to invoker;
  2. complete attachments;
  3. merge master;
  4. upgrade hessian2

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

### CODE REVIEW

File: common/constant/env.go [hunk 1]
```diff
@@ -21,4 +21,6 @@ const (
 	CONF_CONSUMER_FILE_PATH        = "CONF_CONSUMER_FILE_PATH"
 	CONF_PROVIDER_FILE_PATH        = "CONF_PROVIDER_FILE_PATH"
 	APP_LOG_CONF_FILE       string = "APP_LOG_CONF_FILE"
+	DUBBO_VERSION_KEY              = "dubbo"
+	DUBBO_VERSION                  = "2.0.2"
```
Comment (AlexStocks) [inline common/constant/env.go]: "2.0.2" dubbo version? does it exist?

File: common/proxy/proxy_factory/default.go [hunk 2]
```diff
@@ -51,6 +61,86 @@ func (factory *DefaultProxyFactory) GetProxy(invoker protocol.Invoker, url *comm
...
+
+	methodName := invocation.MethodName()
+	proto := url.Protocol
+	path := strings.TrimPrefix(url.Path, "/")
+	args := invocation.Arguments()
+
+	// get service
+	svc := common.ServiceMap.GetService(proto, path)
+	if svc == nil {
+		logger.Errorf("cannot find service [%s] in %s", path, proto)
```
Comment (AlexStocks) [inline common/proxy/proxy_factory/default.go]: add "[ProxyInvoker][Invoke] as log prefix.

File: common/proxy/proxy_factory/default.go [hunk 3]
```diff
@@ -51,6 +61,86 @@ func (factory *DefaultProxyFactory) GetProxy(invoker protocol.Invoker, url *comm
...
+		result.SetError(perrors.Errorf("cannot find service [%s] in %s", path, proto))
+		return result
+	}
+
+	// get method
+	method := svc.Method()[methodName]
+	if method == nil {
+		logger.Errorf("cannot find method [%s] of service [%s] in %s", methodName, path, proto)
```
Comment (AlexStocks) [inline common/proxy/proxy_factory/default.go]: add "[ProxyInvoker][Invoke] as log prefix.

File: protocol/dubbo/readwriter.go [hunk 4]
```diff
@@ -138,14 +138,14 @@ func (p *RpcServerPackageHandler) Read(ss getty.Session, data []byte) (interface
...
 			}
 			if req[6] != nil {
-				attachments = req[6].(map[interface{}]interface{})
+				attachments = req[6].(map[string]string)
 			}
-			pkg.Service.Interface = attachments[constant.INTERFACE_KEY].(string)
-			if pkg.Service.Path == "" && attachments[constant.PATH_KEY] != nil {
-				pkg.Service.Path = attachments[constant.PATH_KEY].(string)
+			pkg.Service.Interface = attachments[constant.INTERFACE_KEY]
+			if pkg.Service.Path == "" && attachments[constant.PATH_KEY] != "" {
```
Comment (AlexStocks) [inline protocol/dubbo/readwriter.go]: use len(xx) != 0 pls.

File: protocol/dubbo/readwriter.go [hunk 5]
```diff
@@ -138,14 +138,14 @@ func (p *RpcServerPackageHandler) Read(ss getty.Session, data []byte) (interface
...
+				pkg.Service.Path = attachments[constant.PATH_KEY]
 			}
-			if attachments[constant.GROUP_KEY] != nil {
-				pkg.Service.Group = attachments[constant.GROUP_KEY].(string)
+			if attachments[constant.GROUP_KEY] != "" {
```
Comment (AlexStocks) [inline protocol/dubbo/readwriter.go]: use len(xxx) != 0 pls.

File: registry/protocol/protocol.go [hunk 6]
```diff
@@ -18,6 +18,7 @@
 package protocol
 
 import (
+	"github.com/apache/dubbo-go/common/proxy/proxy_factory"
```
Comment (AlexStocks) [inline registry/protocol/protocol.go]: another import block.

File: common/proxy/proxy_factory/default.go (hunk 2 above)
Comment (fangyincheng) [reply common/proxy/proxy_factory/default.go]: logger will show the stack

File: common/proxy/proxy_factory/default.go (hunk 3 above)
Comment (fangyincheng) [reply common/proxy/proxy_factory/default.go]: logger will show the stack

File: common/constant/env.go (hunk 1 above)
Comment (fangyincheng) [reply common/constant/env.go]: `response attachmentds` will be validated whether it is supported in dubbo server. And when dubbo protocol version is `2.0.10~2.62`, it is not supported. Refer to https://github.com/apache/dubbo/blob/a36cc7520e0150d07d30e8baa0c61d5f9f11e6ed/dubbo-common/src/main/java/org/apache/dubbo/common/Version.java#L96

File: protocol/dubbo/readwriter.go (hunk 5 above)
Comment (fangyincheng) [reply protocol/dubbo/readwriter.go]: done.

File: protocol/dubbo/readwriter.go (hunk 4 above)
Comment (fangyincheng) [reply protocol/dubbo/readwriter.go]: done.

File: common/constant/env.go (hunk 1 above)
Comment (AlexStocks) [reply common/constant/env.go]: Maybe you should place it into the code as a comment.

Comment (fangyincheng) [reply common/constant/env.go]: yes, I am doing it.

**Pull request merged**